include/core/shape.h

```diff
@@ -510,25 +596,361 @@ class CompositeShape : public Shape<dim>
+
+private:
+  std::vector<std::shared_ptr<Shape<dim>>> components;
+};
+
+
+/**
+ * @tparam dim Dimension of the shape
+ * @class RBF Shapes express the signed distance function as a linear
+ * combination of Radial Basis Functions, which have a defined support radius
+ * and basis function. A collection of nodes and weights compose the object.
+ * Outside of the domain covered by the nodes, the distance is computed by using
+ * the distance to a bounding box instead.
+ */
+template <int dim>
+class RBFShape : public Shape<dim>
+{
+public:
+  /**
+   * Class taken from Optimad Bitpit. https://github.com/optimad/bitpit
+   * @enum RBFBasisFunction
+   * @brief Enum class defining types of RBF kernel functions that could be used
+   * in the class
+   */
+  enum class RBFBasisFunction : int
+  {
+    CUSTOM,
+    WENDLANDC2,
+    LINEAR,
+    GAUSS90,
+    GAUSS95,
+    GAUSS99,
+    C1C0,
+    C2C0,
+    C0C1,
+    C1C1,
+    C2C1,
+    C0C2,
+    C1C2,
+    C2C2,
+  };
+
+  /**
+   * @brief An RBFShape represents a physical object by describing its signed
+   * distance field with a linear combination of radial basis functions. Each
+   * radial basis function has a location and properties that are used in the
+   * sum.
+   * @param support_radius the scaling of the reach of the nodes
+   * @param basis_function the basis function that was used to parametrize the RBF object
+   * @param weight the weighting associated to each node for the sum operation
+   * @param nodes the center of each basis function
+   * @param position the location of the RBF shape
+   * @param orientation the orientation of the shape in relation to each main
+   * axis
+   */
+  RBFShape<dim>(const std::vector<double> &          support_radii,
+                const std::vector<RBFBasisFunction> &basis_functions,
+                const std::vector<double> &          weights,
+                const std::vector<Point<dim>> &      nodes,
+                const Point<dim> &                   position,
+                const Tensor<1, 3> &                 orientation)
+    : Shape<dim>(support_radii[0], position, orientation)
+    , number_of_nodes(weights.size())
+    , weights(weights)
+    , nodes(nodes)
+    , support_radii(support_radii)
+    , basis_functions(basis_functions)
+  {
+    initialize_bounding_box();
+  }
+
+  /**
+   * @brief An RBFShape represents a physical object by describing its signed
+   * distance field with a linear combination of radial basis functions. Each
+   * radial basis function has a location and properties that are used in the
+   * sum.
+   * @param shape_arguments the concatenated vector of all shape arguments for
+   * an RBF in the order: weights, support_radii, basis_functions, nodes_x,
+   * nodes_y, nodes_z
+   * @param position the location of the RBF shape
+   * @param orientation the orientation of the shape in relation to each main
+   * axis
+   */
+  RBFShape<dim>(const std::vector<double> &shape_arguments,
```

Comment:
I would implement the constructor in the .cc file

Reply:
Do you mean the 2nd constructor only or both of them?

Reply:
I'd say both